File: backend/src/main/java/com/project/dto/professional/request/ProfessionalAvailabilityDTO.java
package com.project.dto.professional.request;

import java.time.LocalTime;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.project.enums.DayOfWeekEnum;

import lombok.Data;

@Data
public class ProfessionalAvailabilityDTO {

    private DayOfWeekEnum dayOfWeek;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "HH:mm")
    private LocalTime startTime;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "HH:mm")
    private LocalTime endTime;
}

File: backend_dotnet/SafeMind.Core/DTOs/Professional/Request/ProfessionalAvailabilityDTO.cs
using SafeMind.Core.Enums;

namespace SafeMind.Core.DTOs.Professional.Request
{
    public class ProfessionalAvailabilityDTO
    {
        public DayOfWeekEnum DayOfWeek { get; set; }
        // Using string "HH:mm" for input parsing simplicity, or TimeSpan
        // Java used LocalTime with JsonFormat.
        // We'll use string and parse it in Service or Controller, or TimeSpan directly.
        // Let's use string to match "HH:mm" exactly.
        public string StartTime { get; set; } = string.Empty; 
        public string EndTime { get; set; } = string.Empty;
    }
}
